controllers/cluster_controller_phases.go

```diff
@@ -206,6 +214,19 @@ func (r *ClusterReconciler) reconcileControlPlane(ctx context.Context, cluster *
 		return nil
```

Comment:
It looks like we'll return early here if the control plane resource is being deleted. I'm wondering if we need to handle updating conditions in that case as well.

Reply:
same as above